core/templates/pages/topic-viewer-page/practice-tab/practice-tab.component.ts

```diff
@@ -95,6 +105,45 @@ export class PracticeTabComponent implements OnInit {
       this.classroomUrlFragment = (
         this.urlService.getClassroomUrlFragmentFromLearnerUrl());
     }
+    this.topicViewerBackendApiService.fetchTopicDataAsync(
+      this.topicUrlFragment, this.classroomUrlFragment).then(
+      (readOnlyTopic: ReadOnlyTopic) => {
+        this.topicNameTranslationKey =
+          this.i18nLanguageCodeService.getTopicTranslationKey(
+            readOnlyTopic.getTopicId(), TranslationKeyType.TITLE
+          );
+        this.obtainTranslatedTopicName();
+        this.subscribeToOnLangChange();
+      });
+  }
+
+  ngOnDestroy(): void {
+    this.directiveSubscriptions.unsubscribe();
+  }
+
+  subscribeToOnLangChange(): void {
+    this.directiveSubscriptions.add(
+      this.translateService.onLangChange.subscribe(() => {
+        this.obtainTranslatedTopicName();
+      })
+    );
+  }
+
+  obtainTranslatedTopicName(): void {
+    if (this.shouldGetTranslatedTopicName()) {
+      this.translatedTopicName = this.translateService.instant(
+        this.topicNameTranslationKey);
+    } else {
+      this.translatedTopicName = this.topicName;
+    }
+  }
+
+  shouldGetTranslatedTopicName(): boolean {
```

Comment:
Rename this function to something like ‘isTranslationAvailable’ or something like that.

Reply:
Done